src/main/scala/com/datastax/gatling/plugin/utils/CqlPreparedStatementUtil.scala

```diff
@@ -44,12 +45,80 @@ trait CqlPreparedStatementUtil {
   def getParamsList(preparedStatement: PreparedStatement): List[Int]
 }
 
+object SessionCollectionResolver {
+  def get(session:Session, name:String):Option[Any] = session.attributes.get(name)
+
+  def getClz[T <: Any](session:Session, name:String):Option[Class[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case clz:Class[T] => Option(clz)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getIterable[T <: Any](session:Session, name:String):Option[lang.Iterable[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Iterable[T] => Option(rv.asJava)
```

Comment:
I feel that all conversions should be another way around.
We want to convert java code to scala ecosystem API.
So if this is scala it should be propagated as scala:
` case rv:Iterable[T] => Option(rv)`
and if it is java then it should be converted to scala:
`case rv:lang.Iterable[T] => Option(rv.asScala)`
(same for `getMap()`) Why we are not doing it?

Also, please simplify by returning `Iterable[T]` (or `Iterator[T]`) instead of `Option[Iterable[T]]`

Reply:
In general I agree with you (at least about the Scala Iterable part), but this is something of an unusual case.  Note that we use this method to get objects which will subsequently be used as params to a Java API call.  I _could_ return a Scala Iterator here, but I'd wind up immediately converting it back into a Java Iterator... which seemed counter-productive.

I'll add some comments to that effect to make the relationship clearer.

I disagree about avoiding the use of Option.  We're pulling values out of a Gatling session here and that Session may or may not have a value with the input name in it.  We need to clearly distinguish cases in which that val just isn't defined in the Session from all other cases.  The Option type makes that very explicit.

Reply:
Bah, I was wrong about the Iterable stuff... I got this confused with the asList/Set/Map functions elsewhere.  These methods are just used for the case where we need to pull a class out of a collection, and in that case I completely agree with your statements above re: use of the Scala types.  Converting to that now.

I still don't agree about the Option, though. :)